### ISSUE TITLE
Using callback parameter

### ISSUE THREAD

**Issue by stealth35** [AOP-PHP/AOP#10]
For all `aop_add_around` the callback parameter wait a zval (`z`) it's should be directly a calllback (`f`)

> ```
> f - function or array containing php method call info (returned as
>     zend_fcall_info and zend_fcall_info_cache)
> ```
> 
> But something wrong in `zend_call_function` the callack is transformed to an empty `array`
#### Example  :

``` c
PHP_FUNCTION(aop_add_before)
{
    char *selector;
    int selector_len;
    zend_fcall_info fci;
    zend_fcall_info_cache fcic;

    if (zend_parse_parameters(ZEND_NUM_ARGS() TSRMLS_CC, "sf", &selector, &selector_len, &fci, &fcic) == FAILURE) {
        return;
    }

    add_pointcut(fci, fcic, selector, selector_len, AOP_KIND_BEFORE TSRMLS_CC);
}

static void add_pointcut (zend_fcall_info fci, zend_fcall_info_cache fcic, char *selector, int selector_len, int type TSRMLS_DC) {
    pointcut *pc;
    int count;

    aop_g(count_pcs)++;
    count=aop_g(count_pcs)-1;

    if (aop_g(count_pcs)==1) {
        aop_g(pcs) = emalloc(sizeof(pointcut *));
    } else {
        aop_g(pcs) = erealloc(aop_g(pcs),aop_g(count_pcs)*sizeof(pointcut *));
    }

    pc = emalloc(sizeof(pointcut));
    pc->selector = selector;
    pc->fci = fci;
    pc->fcic = fcic; 
    pc->kind_of_advice = type;
    parse_pointcut(&pc);
    aop_g(pcs)[count] = pc;
}
```

**Comment by geraldcroes** [comment]
Already fixed

**Comment by stealth35** [comment]
Yeah it's fix in 8a26642 sorry, I missed the `Z_ADDREF_P`
